### PR TITLE
Refactor: Enrollment success logout

### LINKED ISSUE #466: Update Enrollment Success page with sign out button / text
image

### PR DESCRIPTION
**Pull request by machikoyasuda**
Closes #466

## What this PR does

- [x] - If auth is required and the user is logged in, show the Login.gov paragraph with button
- [x] - If auth is required and user is logged out, don't show Login.gov paragraph with button
- [x] - Add `Button.logout`, with fallback text but without a label
- [x] - Style button for this page
- [x] - Copy: Add paragraph, both parts to English and Spanish locale files
- Note on tests: Using `visit()` cleared out cookies, b/c of a Cypress bug. Used `cy.window.location.href` instead https://github.com/cypress-io/cypress/issues/18690

## Flows to test for
- If auth is required and the user is logged in, show the Login.gov paragraph with button
- If auth is required and user is logged out, don't show Login.gov paragraph with button
- If auth is not required, don't show paragraph ever
- Mobile views

### Enrollment success
image
image

### Logout success
image
image

### CODE REVIEW

**Comment by machikoyasuda** [comment]
@thekaveman @angela-tran Ready for review

**Comment by angela-tran** [comment]
Reviewing this now! 👀

**Review by angela-tran — DISMISSED** [review]
I have some minor notes, but I don't think they should hold up merging this in. It's really just cleanup that can be done separately:
 - I think it'd make sense to have `Button.login()` and `Button.logout()` as separate methods
 - There is some duplication we could consolidate in the way we create the `Page`s in `success()`

The pages look great!! 🔥 
image

image

**Comment by machikoyasuda** [comment]
> I have some minor notes, but I don't think they should hold up merging this in. It's really just cleanup that can be done separately:
> 
>     * I think it'd make sense to have `Button.login()` and `Button.logout()` as separate methods
> 

This is a good idea! Going to implement it in this PR.

**Comment by angela-tran** [comment]
Can you move the `url` and `text` into `Button.login` as well? I think that would nicely finish out the conversation started in https://github.com/cal-itp/benefits/pull/470/files/850eaf1421a8ed9f25069a023798df3e74d6cbeb#r859218184

**Comment by machikoyasuda** [comment]
> Can you move the `url` and `text` into `Button.login` as well? I think that would nicely finish out the conversation started in https://github.com/cal-itp/benefits/pull/470/files/850eaf1421a8ed9f25069a023798df3e74d6cbeb#r859218184

Will do this in next PR for this issue: https://github.com/cal-itp/benefits/issues/507

**Comment by machikoyasuda** [comment]
@angela-tran Investigating test failures

File: benefits/enrollment/views.py [hunk 1]
```diff
@@ -156,10 +156,7 @@ def success(request):
             raise Exception("EligibilityVerifier requires authentication, but OAUTH_CLIENT_NAME is None")
 
         if session.logged_in(request):
-            button = viewmodels.Button.login(
-                classes="logout",
```
Comment (angela-tran) [inline benefits/enrollment/views.py]: Oh, `classes` is missing from the code in the method

Comment (machikoyasuda) [reply benefits/enrollment/views.py]: Yeah - Ended up removing it b/c I am not using it for any CSS actually, and now the button is more clear b/c the method is now named "logout()".

**Comment by machikoyasuda** [comment]
Just realized the UI tests never ran on this PR until the previous PR!

Now that the Enrollment Success page has logic checking whether a user is logged in to Login.gov or not, the Cypress tests cannot adequately test the AuthProvider flow -- because we can't log a user in through Login.gov with our Cypress tests. I have added a `skip()` on the specs for now (but maybe I should remove it entirely(??))

cc @thekaveman

**Comment by angela-tran** [comment]
The video shows that the test expects to see "Success!...", but it ends up seeing the "logged-out" page. This makes sense since the UI test can't actually go through the login process.

image

File: tests/cypress/specs/ui/enrollment.spec.js [hunk 2]
```diff
@@ -7,10 +7,12 @@ describe("Single Verifier, No AuthProvider: Enrollment success page", () => {
     cy.contains(agencies[1].fields.short_name).click();
     cy.contains("Get started").click();
     cy.contains("Continue").click();
-    cy.visit("/enrollment/success");
+    cy.url().should("include", eligibility_confirm_url);
```
Comment (thekaveman) [inline tests/cypress/specs/ui/enrollment.spec.js:12]: Need to define this variable

File: tests/cypress/specs/ui/enrollment.spec.js [hunk 3]
```diff
@@ -7,10 +7,12 @@ describe("Single Verifier, No AuthProvider: Enrollment success page", () => {
...
   });
 
   it("Has the appropriate English copy for all flows", () => {
+    cy.visit("/enrollment/success");
```
Comment (thekaveman) [inline tests/cypress/specs/ui/enrollment.spec.js:16]: I think this line is needed in the Spanish test too, right?

**Review by thekaveman — CHANGES_REQUESTED** [review]

File: tests/cypress/specs/ui/enrollment.spec.js [hunk 4]
```diff
@@ -1,13 +1,16 @@
 const agencies = require("../../fixtures/transit-agencies");
 
+const eligibility_confirm_url = "/eligibility/confirm";
```
Comment (angela-tran) [inline tests/cypress/specs/ui/enrollment.spec.js:3]: I noticed you removed usage of this variable in https://github.com/cal-itp/benefits/pull/493/commits/7c5e83eff093ec84ba366b44e06c698661e3d616, so we should remove it now, right?

```suggestion
```

**Comment by angela-tran** [comment]
@machikoyasuda I'll fix that minor pre-commit error since I overlooked the extra whitespace

**Review by thekaveman — APPROVED** [review]
This is great! I opened #513 as a small refactor to-do later.

✅ As a logged in user, clicked the sign out button, and was shown the "You have been logged out" message with image.

✅ As an anonymous user where auth was not required, was not shown the paragraph / button to sign out.

**Review by angela-tran — APPROVED** [review]

**Pull request merged**